tests/functional/includes/date.mock.js

```diff
@@ -1,14 +1,30 @@
-// Highjack the Date object so that our tests don't suddenly start failing...
+(function () {
+    'use strict';
 
-var __Date__ = Date;
+    // Create a date object for a specific instant in time (my 36th birthday...)
 
-// February 13, 2015
-var STATIC_DATE_MS = 1423852666565;
+    var d = new Date();
+    d.setUTCFullYear(2014, 11, 8);
+    d.setUTCHours(12, 34, 0, 0);
 
-Date = function (value) {
-    return new __Date__(value !== undefined ? value : STATIC_DATE_MS);
-};
+    // Adjust it based on the time zone offset, so that formatted date/time
+    // will be what we expect, independent of the time zone (minus the time
+    // zone name, which is a different story)
 
-Date.now = function () {
-    return STATIC_DATE_MS;
-};
+    d.setUTCMinutes(d.getUTCMinutes() + d.getTimezoneOffset());
```

Comment:
Yeah I understand how it increments the other values — seems like a great approach. I was just wondering why they used the word "attempts" which makes it seem like it might not always work.